src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php

```diff
@@ -737,11 +758,47 @@ public function iShouldBeInformedThatCartIsNoLongerAvailable(): void
         Assert::same($this->responseChecker->getResponseContent($response)['message'], 'Not Found');
     }
 
-    private function isViolationWithMessageInResponse(Response $response, string $message): bool
+    /**
+     * @Then I should not be able to specify province name manually for shipping address
+     */
+    public function iShouldNotBeAbleToSpecifyProvinceNameManuallyForShippingAddress(): void
+    {
+        $this->iCompleteTheAddressingStep();
+
+        /** @var Response $response */
+        $response = $this->client->getResponse();
+        Assert::same($response->getStatusCode(), 400);
+        Assert::true($this->isViolationWithMessageInResponse(
+            $response,
+            'Please select proper province.',
+            'shippingAddress'
+        ));
+    }
+
+    /**
+     * @Then I should not be able to specify province name manually for billing address
+     */
+    public function iShouldNotBeAbleToSpecifyProvinceNameManuallyForBillingAddress(): void
+    {
+        /** @var Response $response */
+        $response = $this->client->getResponse();
+        Assert::same($response->getStatusCode(), 400);
+        Assert::true($this->isViolationWithMessageInResponse(
+            $response,
+            'Please select proper province.',
+            'billingAddress'
+        ));
+    }
```

Comment:
```suggestion
    /**
     * @Then /^I should not be able to specify province name manually for (billing|shipping) address$/
     */
    public function iShouldNotBeAbleToSpecifyProvinceNameManuallyForAddress(string addressType): void
    {
        /** @var Response $response */
        $response = $this->client->getResponse();
        Assert::same($response->getStatusCode(), 400);
        Assert::true($this->isViolationWithMessageInResponse(
            $response,
            'Please select proper province.',
            $addressType . 'Address'
        ));
    }
```
You can merge this two steps to one

Reply:
Not completely, because of line 766, I could extract asserts to a private method